lib/controllers/hosted-skill-controller/helper.js

```diff
@@ -81,3 +95,64 @@ function handleSkillStatus(response, skillId, callback) {
         callback(`${errMessage}${errReason}Infrastructure provision for the hosted skill failed. Please try again.`);
     }
 }
+
+function downloadAuthInfoScript(callback) {
+    const authInfoUrl = CONSTANTS.HOSTED_SKILL.S3_SCRIPTS.AUTH_INFO.URL;
+    const authInfoPath = path.join(os.homedir(),
+        CONSTANTS.FILE_PATH.ASK.HIDDEN_FOLDER,
+        CONSTANTS.FILE_PATH.ASK.AUTH_INFO);
+    this.downloadScriptFromS3(authInfoUrl, authInfoPath, (err) => {
+        if (err) {
+            return callback(err);
+        }
+        return callback(null);
+    });
+}
+
+function downloadAskPrePushScript(callback) {
+    const askScriptUrl = CONSTANTS.HOSTED_SKILL.S3_SCRIPTS.ASK_PRE_PUSH.URL;
+    const scriptFolderPath = path.join(os.homedir(),
+        CONSTANTS.FILE_PATH.ASK.HIDDEN_FOLDER,
+        CONSTANTS.FILE_PATH.ASK.SCRIPTS_FOLDER.NAME);
+    const askFilePath = path.join(os.homedir(),
+        CONSTANTS.FILE_PATH.ASK.HIDDEN_FOLDER,
+        CONSTANTS.FILE_PATH.ASK.SCRIPTS_FOLDER.NAME,
+        CONSTANTS.FILE_PATH.ASK.SCRIPTS_FOLDER.ASK_PRE_PUSH);
+    if (!fs.existsSync(scriptFolderPath)) {
+        fs.mkdirSync(scriptFolderPath);
+    }
+    this.downloadScriptFromS3(askScriptUrl, askFilePath, (err) => callback(err || null));
+}
+
+function downloadGitCredentialHelperScript(callback) {
+    const credentialScriptUrl = CONSTANTS.HOSTED_SKILL.S3_SCRIPTS.GIT_CREDENTIAL_HELPER.URL;
+    const scriptFolderPath = path.join(os.homedir(),
+        CONSTANTS.FILE_PATH.ASK.HIDDEN_FOLDER,
+        CONSTANTS.FILE_PATH.ASK.SCRIPTS_FOLDER.NAME);
+    const credentialFilePath = path.join(os.homedir(),
+        CONSTANTS.FILE_PATH.ASK.HIDDEN_FOLDER,
+        CONSTANTS.FILE_PATH.ASK.SCRIPTS_FOLDER.NAME,
+        CONSTANTS.FILE_PATH.ASK.SCRIPTS_FOLDER.GIT_CREDENTIAL_HELPER);
+    if (!fs.existsSync(scriptFolderPath)) {
+        fs.mkdirSync(scriptFolderPath);
+    }
+    this.downloadScriptFromS3(credentialScriptUrl, credentialFilePath, (err) => callback(err || null));
+}
+
+function downloadScriptFromS3(scriptUrl, filePath, callback) {
+    const DOWNLOADED = 'downloaded';
+    const retryCall = (loopCallback) => {
+        const writeStream = fs.createWriteStream(filePath);
+        https.get(scriptUrl, (res) => {
+            res.pipe(writeStream);
+            writeStream.on('finish', () => {
+                fs.chmodSync(filePath, '777');
```

Comment:
I think 777 is too open?

Reply:
Do you think we need different chmod for each script? Like, 711 for the auth_info and 755 for the rest?

Reply:
I think 700